### DISCUSSION TITLE
T2: Ranking PRELIMINAR bonus

### DISCUSSION THREAD

**Discussion by benjavicente** [IIC2133-PUC/2024-1#153] [Importante]
# Rankings

Este es el ranking **preliminar** del bonus de la T2.

- Se compila y corre los programas en un MacOS M1 para los **tests hard-3**.
- Las entregas que **no compilan o que tenga parte incorrecta** (se cae o no tiene el output esperado), **no se muestran en el ranking**.
- Solo se muestran las personas que rellenaron el formulario.
- Que aparezcas aquí **NO** quiere decir que tengas un 7 (todos los tests buenos). El ranking de tiempo no mide la complejidad. No cumplir la complejidad (solucionarlo a fuerza bruta) se considera como parte inválida (un 1).
- Al final, **se ignorará si se respondió T1 o T3**. Se verificará automáticamente si corre para tomarlo en el ranking.

> [!WARNING]
> Como se menciona antes, el código preliminar se ejecuta en MacOS.
> Si en su computador compila y aquí les sale que no, vean #165.

> [!TIP]
> Si quieren ver en que se demora el código, pueden usar [flamegraph](https://github.com/flamegraph-rs/flamegraph).

Última actualización: 14:30 - 31/May

## 1. find_sequences
| | User | Median |
| - |------|--------|
| 1 | GithubSPerez | 0.00601s |
| 2 | c4ebt | 0.00903s |
| 3 | hansjkl | 0.00948s |
| 4 | gbarrueto | 0.01017s |
| 5 | mhelmm | 0.01039s |
| 6 | Segismundios | 0.01080s |
| 7 | Cristian1754 | 0.01114s |
| 8 | dedmu5 | 0.01132s |
| 9 | NicolasOlmosQuiroga | 0.01230s |
| 10 | catalinacarter | 0.01296s |
| 11 | tsunayoshi21 | 0.01299s |
| 12 | lfb1206 | 0.01305s |
| 13 | ManuelV03 | 0.01401s |
| 14 | MaxAl100 | 0.01557s |
| 15 | ALittleChameleon | 0.01658s |
| 16 | Juan-M-Ovalle | 0.01912s |
| 17 | vgalfaro | 0.01925s |
| 18 | VicentePareja | 0.02193s |
| 19 | gaspar-mr05 | 0.02360s |
| 20 | Pedr0sit0s | 0.02667s |
| 21 | ResistorCat | 0.02779s |
| 22 | FradexMF | 0.03194s |
| 23 | vicenteCordero | 0.04061s |
| 24 | jimmygallardo | 0.06213s |
| 25 | RadiosMark | 0.07042s |
| 26 | gabo1243 | 0.18917s |
| 27 | JavierLarre | 0.21922s |
| 28 | IanZmen | 0.26036s |
| 29 | Monti7475 | 0.31386s |
| 30 | sofialarrain | 0.31415s |
| 31 | csilval | 0.31862s |
| 32 | joaquindm02 | 0.32064s |
| 33 | ljgonzalez1 | 0.32224s |
| 34 | tsbravo | 0.33596s |
| 35 | JaviPeace | 0.35469s |
| 36 | mhmveloper | 0.42401s |
| 37 | DiegoVarela11 | 0.62105s |
| 38 | Consolo0 | 0.91752s |
| 39 | Fernanda-Bley | 2.14250s |
| 40 | Joako5848 | 2.70961s |
| 41 | amundaca02 | 4.58996s |
| 42 | natydlb | 4.71726s |
| 43 | isufan | 5.25302s |
| 44 | cattpuccino | 5.50441s |
| 45 | AleSierra26 | 5.80808s |
| 46 | Gcamposbriceno | 5.81531s |
| 47 | jalonsoluc | 6.09799s |
- Trycoz: COMPILE ERROR
- ssilvapinochet: COMPILE ERROR
- Tariztia: INCORRECT OUTPUT
- JpOlivares08: RUNTIME ERROR
- leonelcespedess: INCORRECT OUTPUT
- CarlangaUC: COMPILE ERROR
- tariztia: INCORRECT OUTPUT

## 3. patterns
| | User | Median |
| - |------|--------|
| 1 | gbarrueto | 0.05316s |
| 2 | c4ebt | 0.10404s |
| 3 | jalonsoluc | 0.10765s |
| 4 | mhelmm | 0.12530s |
| 5 | Juan-M-Ovalle | 0.13848s |
| 6 | tsbravo | 0.18810s |
| 7 | gabo1243 | 0.25423s |
| 8 | joaquindm02 | 0.43735s |
| 9 | tariztia | 0.96181s |
| 10 | Tariztia | 0.97180s |
| 11 | Consolo0 | 1.00257s |
| 12 | ResistorCat | 1.10439s |
| 13 | tsunayoshi21 | 2.04236s |
| 14 | hansjkl | 6.67254s |
| 15 | RadiosMark | 6.75856s |
- AleSierra26: INCORRECT OUTPUT
- VicentePareja: TIMEOUT
- MaxAl100: TIMEOUT
- JavierLarre: INCORRECT OUTPUT
- Trycoz: COMPILE ERROR
- Fernanda-Bley: INCORRECT OUTPUT
- Cristian1754: INCORRECT OUTPUT
- cattpuccino: INCORRECT OUTPUT
- vgalfaro: INCORRECT OUTPUT
- sofialarrain: INCORRECT OUTPUT
- ALittleChameleon: INCORRECT OUTPUT
- Monti7475: INCORRECT OUTPUT
- JpOlivares08: RUNTIME ERROR
- ManuelV03: TIMEOUT
- Gcamposbriceno: INCORRECT OUTPUT
- dedmu5: RUNTIME ERROR
- leonelcespedess: INCORRECT OUTPUT
- GithubSPerez: TIMEOUT
- natydlb: TIMEOUT

**Comment by MaxAl100** [comment]
El ranking actual significa que nadie de los 7 inscritos ha resuelto el problema 3? O que no son suficientemente rápidos para aparecer en el ranking?
Porque mi código corre el hard-3 bien, pero no lo he optimizado (se demora 30 segundos 😅 )

**Reply by benjavicente** [reply]
No son lo suficientemente rápidos para pasar el test hard en menos de 10s 😅

**Reply by MaxAl100** [reply]
aprovechando que fue hace 1 minuto, cada cuanto actualizan los rankings?

**Reply by benjavicente** [reply]
Los estoy revisando principalmente cuando hay nuevas respuestas en el formulario y en cada tarde.

**Comment by Cristian1754** [comment]
Hola, ¿de que tamaño es la muestra para determinar la mediana del codigo? Me surge la duda ya que he notado que en cada actualizaciòn del ranking mi tiempo varia mucho incluso cuando es el mismo commit el que se evalua. Podria ser posible aumentar esta muestra? De tal manera que las posiciones sean mas representativas y no caigan en un factor de azar.

**Reply by benjavicente** [reply]
La cantidad de muestras es variable dependiendo del tiempo de ejecución.

Uso [hyperfine](https://github.com/sharkdp/hyperfine) para obtener los tiempos, con 1 corrida de warmup, 3 corridas mínimas y samples por al menos 3 segundos.

Códigos rápidos alcanzan a correr sobre 100 veces (el tuyo corre como 150) y las lentas están entre 3 y 5, lo que igual, al ser más lentas, dan un tiempo más certero.

Obviamente hay factores externos. Puede que un código corra levemente más rápido porque otro programa de mi PC decidió ocupar más recursos mientras corría un test (y variaciones de aprox 5ms hacen que no sea tan estable como dices).

El ranking **real** será ejecutado con más cuidado con esos factores externos.

**Comment by GithubSPerez** [comment]
Hola, sabes que rellené el form antes de la última actualización y no veo mi resultado y tampoco veo cual fue el error, esto significa que mi código no compiló correctamente?

**Reply by benjavicente** [reply]
No. Por alguna razón no fue recolectado tu usuario cuando realicé esa actualización.

**Reply by GithubSPerez** [reply]
a xd
